zson/marshal.go

```diff
@@ -292,8 +293,10 @@ func (m *MarshalZNGContext) encodeValue(v reflect.Value) (zng.Type, error) {
 		kind := v.Kind().String()
 		if name != "" && name != kind {
 			// We do not want to further decorate nano.Ts as
-			// it's already been converted to a Zed time.
-			if v.Type() == nanoTsType {
+			// it's already been converted to a Zed time;
+			// likewise for zng.Value, which gets encoded as
+			// itself and its own alias type if it has one.
+			if v.Type() == nanoTsType || v.Type() == zngValueType {
```

Comment:
Nit: Call `v.Type` once.
```suggestion
			if t := v.Type(); t == nanoTsType || t == zngValueType {
```